fuel-p2p/src/orchestrator.rs

```diff
@@ -0,0 +1,140 @@
+                                    },
+                                    GossipsubMessage::ConensusVote(vote) => {
+                                        let _ = self.tx_consensus.send(ConsensusBroadcast::NewVote(vote));
+                                    },
+                                }
+                            },
+                            FuelBehaviourEvent::RequestMessage { request_message, request_id } => {
+                                match request_message {
+                                    RequestMessage::RequestBlock(block_height) => {
+                                        let db = self.db.clone();
+
+                                        self.outbound_responses.push(
+                                            Box::pin(async move {
+                                                db.get_sealed_block(block_height).await.map(|block| (OutboundResponse::ResponseBlock(block), request_id))
+                                            })
+                                        );
+                                    }
+                                }
+
+                            },
+                            _ => {}
+                        }
+                    }
+                },
+                module_request_msg = self.rx_request_event.recv() => {
+                    if let Some(request_event) = module_request_msg {
+                        match request_event {
+                            P2pRequestEvent::RequestBlock { height, response } => {
+                                let request_msg = RequestMessage::RequestBlock(height);
+                                let channel_item = ResponseChannelItem::ResponseBlock(response);
+                                let _ = self.p2p_service.send_request_msg(None, request_msg, channel_item);
+                            },
+                            P2pRequestEvent::BroadcastNewBlock { block } => {
+                                let broadcast = GossipsubBroadcastRequest::NewBlock(block);
+                                let _ = self.p2p_service.publish_message(broadcast);
+                            },
+                            P2pRequestEvent::BroadcastNewTransaction { transaction } => {
+                                let broadcast = GossipsubBroadcastRequest::NewTx(transaction);
+                                let _ = self.p2p_service.publish_message(broadcast);
+                            },
+                            P2pRequestEvent::BroadcastConsensusVote { vote } => {
+                                let broadcast = GossipsubBroadcastRequest::ConensusVote(vote);
+                                let _ = self.p2p_service.publish_message(broadcast);
+                            }
+                        }
+                    } else {
+                        warn!(target: "fuel-libp2p", "Failed to receive P2PRequestEvent");
+                    }
+                }
+
+
```

Comment:
normally `cargo fmt` would trim this, but don't think it works inside of select! macros. 
```suggestion
```

Reply:
yes `select!` support is terrible